src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Operations/CosmosFhirOperationsDataStore.cs

```diff
@@ -0,0 +1,205 @@
+{
+    public sealed class CosmosFhirOperationsDataStore : IFhirOperationsDataStore
+    {
+        private readonly Func<IScoped<IDocumentClient>> _documentClientFactory;
+        private readonly IFhirDataStoreContext _fhirDataStoreContext;
+        private readonly RetryExceptionPolicyFactory _retryExceptionPolicyFactory;
+        private readonly ILogger _logger;
+
+        private readonly GetAvailableExportJobs _getAvailableExportJobs = new GetAvailableExportJobs();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CosmosFhirOperationsDataStore"/> class.
+        /// </summary>
+        /// <param name="documentClientFactory">The factory for <see cref="IDocumentClient"/>.</param>
+        /// <param name="fhirDataStoreContext">The data store context.</param>
+        /// <param name="retryExceptionPolicyFactory">The retry exception policy factory.</param>
+        /// <param name="logger">The logger.</param>
+        public CosmosFhirOperationsDataStore(
+            Func<IScoped<IDocumentClient>> documentClientFactory,
+            IFhirDataStoreContext fhirDataStoreContext,
+            RetryExceptionPolicyFactory retryExceptionPolicyFactory,
+            ILogger<CosmosFhirOperationsDataStore> logger)
+        {
+            EnsureArg.IsNotNull(documentClientFactory, nameof(documentClientFactory));
+            EnsureArg.IsNotNull(fhirDataStoreContext, nameof(fhirDataStoreContext));
+            EnsureArg.IsNotNull(retryExceptionPolicyFactory, nameof(retryExceptionPolicyFactory));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _documentClientFactory = documentClientFactory;
+            _fhirDataStoreContext = fhirDataStoreContext;
+            _retryExceptionPolicyFactory = retryExceptionPolicyFactory;
+            _logger = logger;
+        }
+
+        private IDocumentClient DocumentClient => _documentClientFactory().Value;
+
+        public async Task<ExportJobOutcome> CreateExportJobAsync(ExportJobRecord jobRecord, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(jobRecord, nameof(jobRecord));
+
+            var cosmosExportJob = new CosmosExportJobRecordWrapper(jobRecord);
+
+            try
+            {
+                ResourceResponse<Document> result = await DocumentClient.CreateDocumentAsync(
+                    _fhirDataStoreContext.CollectionUri,
+                    cosmosExportJob,
+                    new RequestOptions() { PartitionKey = new PartitionKey(CosmosDbExportConstants.ExportJobPartitionKey) },
+                    disableAutomaticIdGeneration: true,
+                    cancellationToken: cancellationToken);
+
+                return new ExportJobOutcome(jobRecord, WeakETag.FromVersionId(result.Resource.ETag));
+            }
+            catch (DocumentClientException dce)
+            {
+                if (dce.StatusCode == HttpStatusCode.RequestEntityTooLarge)
```

Comment:
Looks like the wrong error code (below as well)

Reply:
Fixed.